server/src/main/java/org/candlepin/model/AsyncJobStatus.java

```diff
@@ -0,0 +1,535 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.model;
+
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.annotations.Type;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.Map;
+import java.util.HashMap;
+
+import javax.persistence.CollectionTable;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.MapKeyColumn;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+
+
+
+/**
+ * Represents the current status for a long-running job.
+ */
+@XmlRootElement
+@XmlAccessorType(XmlAccessType.PROPERTY)
+@Entity
+@Table(name = AsyncJobStatus.DB_TABLE)
+public class AsyncJobStatus extends AbstractHibernateObject {
+
+    /** Name of the table backing this object in the database */
+    public static final String DB_TABLE = "cp2_async_jobs";
+
+    /** Enum of job states; terminal states represent states at which the job will no longer change */
+    public enum JobState {
+        CREATED(false),
+        QUEUED(false),
+        RUNNING(false),
+        COMPLETED(true),
+        FAILED_WITH_RETRY(false),
+        FAILED(true),
+        CANCELED(true);
```

Comment:
Let's add some java doc for each of these describing what they mean.

Reply:
Done. I also rearranged the states a bit so the terminal states were at the bottom and the listing is, generally, more in line with the expected state transitions.